gradio/inputs.py

```diff
@@ -123,6 +123,7 @@ def __init__(
         numeric: Optional[bool] = False,
         type: Optional[str] = "str",
         label: Optional[str] = None,
+        optional: bool = False,
```

Comment:
But it turns out a secondary reason was that every input component includes an `optional` property in its config because the base Component class has the `optional` attribute. However, when we are recreating the input component from its config, some input components don't accept the `optional` parameter. As a fix, I added the `optional` attribute to every input component, even though in many cases it has no effect (which is documented). 

Is there a better way to do this? @aliabid94?